apis/launchers.md

```diff
@@ -63,46 +63,69 @@ As you can see, the method `set_progress` takes a `double` value type and is a r
 
 Actions are specific functions your app can perform without already being open; think of them as alternate and more specific entry points into your app. Actions appear in the context menu of your app icon in the Applications Menu and Dock, and are searchable by name from the Applications Menu.
 
-Adding actions to a `.desktop` file does not involve writing any code or using any external dependencies, though your app needs a way to distinguish between actions, e.g. with command line flags.
+### D-Bus activation
 
-Actions must first be declared in a new `Actions` line in your app's .desktop file. This line should contain a `;` separated list of unique action names:
+Your app needs to support D-Bus activation in order to use actions as entry points. This does not require any changes to the application source code. All that is needed is a service file which is not unlike the `.desktop` file that you are already familiar with. Define a new `service.in` file in the `data` directory:
+
+```ini
+[D-BUS Service]
+Name=com.github.myteam.myapp
+Exec=@exec_name@ --gapplication-service
+```
+
+Notice the `@exec_name@` configurable variable and the `.in` file extension. Here we are using meson's [configuration feature](https://mesonbuild.com/Configuration.html) to specify the name of the executable. To install the service add this to your `meson.build` file:
+
+```coffeescript
+# Install D-Bus service, so that application can be started by D-Bus
+service_conf_data = configuration_data()
+service_conf_data.set('exec_name', get_option('prefix') / get_option('bindir') / meson.project_name())
+
+configure_file(
+    input: 'data' / 'service.in',
+    output: meson.project_name() + '.service',
+    configuration: service_conf_data,
+    install: true,
+    install_dir: get_option('datadir') / 'dbus-1' / 'services'
+)
+```
```

Comment:
We can simplify this quite a bit by just putting in the exec name ourselves. That would be I think better for a tutorial
```suggestion
Your app needs to support D-Bus activation in order to use actions as entry points. This does not require any changes to the application source code. All that is needed is a service file which is not unlike the `.desktop` file that you are already familiar with. Create a new `.service` file in the `data` directory:

```ini
[D-BUS Service]
Name=com.github.myteam.myapp
Exec=com.github.myteam.myapp --gapplication-service
```

To install the service add the following to your `meson.build` file:

```coffeescript
# Install D-Bus service, so that application can be started by D-Bus
install_data(
    'data' / 'myapp.service',
    install_dir: get_option('datadir') / 'dbus-1' / 'services',
    rename: meson.project_name() + '.service',
)
```

Reply:
I wrongly thought we needed the `prefix/bindir` in there, but the non-template version is much better!